src/eureka/S5_lightcurve_fitting/differentiable_models/SinusoidPhaseCurve.py

```diff
@@ -0,0 +1,162 @@
+            self.maxOrder = np.max(orders)
+        else:
+            raise AssertionError('There are no AmpCos or AmpSin parameters to'
+                                 'fit. Either remove sinusoid_pc or add some'
+                                 'AmpCos or AmpSin terms to fit.')
+
+    @property
+    def time(self):
+        """A getter for the time."""
+        return self._time
+
+    @time.setter
+    def time(self, time_array):
+        """A setter for the time."""
+        self._time = time_array
+        if self.starry_model is not None:
+            self.starry_model.time = time_array
+
+    @property
+    def model(self):
+        """A getter for the model."""
+        return self._model
+
+    @model.setter
+    def model(self, model):
+        """A setter for the model."""
+        self._model = model
+        if self.starry_model is not None:
+            self.starry_model.model = model
+
+    def setup(self, **kwargs):
+        super().setup(**kwargs)
+        if self.starry_model is not None:
+            self.starry_model.setup(**kwargs)
+
+    def update(self, newparams, **kwargs):
+        """Update the model with new parameter values.
+
+        Parameters
+        ----------
+        newparams : ndarray
+            New parameter values.
+        **kwargs : dict
+            Additional parameters to pass to
+            eureka.S5_lightcurve_fitting.models.Model.update().
+        """
+        super().update(newparams, **kwargs)
+        if self.starry_model is not None:
+            self.starry_model.update(newparams, **kwargs)
+
+    def eval(self, eval=True, channel=None, **kwargs):
+        """Evaluate the function with the given values.
+
+        Parameters
+        ----------
+        eval : bool; optional
+            If true evaluate the model, otherwise simply compile the model.
+            Defaults to True.
+        channel : int; optional
+            If not None, only consider one of the channels. Defaults to None.
+        **kwargs : dict
+            Must pass in the time array here if not already set.
+
+        Returns
+        -------
+        ndarray
+            The value of the model at the times self.time.
+        """
+        if channel is None:
+            nchan = self.nchan
+            channels = np.arange(nchan)
+        else:
+            nchan = 1
+            channels = [channel, ]
+
+        if eval:
+            lib = np
+            model = self.fit
+            if self.starry_model is not None:
+                systems = self.starry_model.fit.systems
+        else:
+            lib = tt
+            model = self.model
+            if self.starry_model is not None:
+                systems = self.starry_model.systems
+
+        lcfinal = lib.zeros(0)
+        for c in range(nchan):
+            chan = channels[c]
+            phaseVars = lib.ones(len(self.time))
+            
+            # Compute orbital phase
+            # FINDME: for now this assumes a circular orbit
```

Comment:
Is this here for future work?

Reply:
Ah yes, although the fact of the matter is that sinusoidal phase variations only really make sense where the eccentricity is quite small (where the current approximation isn't terrible). The issue here is that you cannot analytically solve for the true anomaly of an eccentric orbit, so I'm not sure that it's possible to do the same thing I did for the numpy-based model in this theano-based model. I'll see what I can do though